### ISSUE TITLE
Oban.Pro.Workers.Batch requires implementing all callbacks

### ISSUE THREAD

**Issue by enewbury** [sorentwo/oban#307]
### Environment

- oban : 2.0.0, oban_pro: 0.3.0
- erlang 22.3.1, elixir 1.10.3-otp-22

### Current Behavior

It appears that the macro makes a direct reference to all functions within the `perform` implementation, so even though the callbacks are optional in the behavior, it fails to compile
```
== Compilation error in file lib/example/prune/jobs/account.ex ==
** (CompileError) lib/oban/pro/workers/batch.ex:87: undefined function handle_attempted/1
```

### Expected Behavior
Compiles when not all batch callbacks are implemented. Probably just need to wrap these in a function_exported?

Thanks for all the great work on this project!

**Comment by sorentwo** [comment]
@enewbury Thanks for another great bug report. This is fixed in the just-released Oban Pro `v0.3.1` 💛

**Comment by enewbury** [comment]
Awesome, thanks for getting that in so quickly!
